addons/hr_recruitment/models/hr_recruitment.py

```diff
@@ -129,9 +129,10 @@ def _default_company_id(self):
     categ_ids = fields.Many2many('hr.applicant.category', string="Tags")
     company_id = fields.Many2one('res.company', "Company", default=_default_company_id)
     user_id = fields.Many2one('res.users', "Responsible", tracking=True, default=lambda self: self.env.uid)
-    date_closed = fields.Datetime("Closed", readonly=True, index=True)
-    date_open = fields.Datetime("Assigned", readonly=True, index=True)
-    date_last_stage_update = fields.Datetime("Last Stage Update", index=True, default=fields.Datetime.now)
+    date_closed = fields.Datetime("Closed", readonly=True, index=True, compute='_compute_date_closed', store=True)
+    date_open = fields.Datetime("Assigned", readonly=True, index=True, compute='_compute_date_open', store=True)
+    date_last_stage_update = fields.Datetime("Last Stage Update", index=True, default=fields.Datetime.now,
+        compute='_compute_date_last_stage_update', store=True)
```

Comment:
* The default could be removed as this is automatically done by the computer
* The field can set as read-only as there is no reason for customer to manually edit that field